src/languageServerApi/languageServerApiManager.ts

```diff
@@ -1,94 +1,114 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT license.
 
-import { commands, Event, Extension, extensions, Uri } from "vscode";
+import { commands, Disposable, Event, Extension, extensions, Uri } from "vscode";
 import { Commands } from "../commands";
+import { Context, ExtensionName } from "../constants";
 import { contextManager } from "../contextManager";
+import { Settings } from "../settings";
+import { syncHandler } from "../syncHandler";
 import { LanguageServerMode } from "./LanguageServerMode";
 
-class LanguageServerApiManager {
+class LanguageServerApiManager implements Disposable {
+    /**
+     * undefined means a legacy version language server
+     * null means the JDT.LS is not activated
+     */
     private serverMode: LanguageServerMode | null | undefined = null;
 
-    public async isStandardServerReady(): Promise<boolean> {
+    private extensionChangeListener: Disposable;
+
+    public async ready(): Promise<boolean> {
         await this.checkServerMode();
-        // undefined serverMode indicates an older version language server
-        if (this.serverMode === undefined) {
-            return true;
-        }
 
-        if (this.serverMode !== LanguageServerMode.Standard) {
+        if (this.serverMode === null || this.serverMode === LanguageServerMode.LightWeight) {
             return false;
         }
 
-        return true;
-    }
-
-    public async isLightWeightMode(): Promise<boolean> {
-        await this.checkServerMode();
-        return this.serverMode === LanguageServerMode.LightWeight;
-    }
-
-    public async awaitSwitchingServerFinished(): Promise<void> {
-        await this.checkServerMode();
         if (this.serverMode === LanguageServerMode.Hybrid) {
             await new Promise<void>((resolve: () => void): void => {
                 extensions.getExtension("redhat.java")!.exports.onDidServerModeChange(resolve);
```

Comment:
is it possible that serverMode changes from Hybrid to Lightweight?

Reply:
No, Currently we only support this state chain: `Lightweight` -> `Hybrid` -> `Standard`

Reply:
Just curious what if I change settings from "hybrid" to "lightweight" during launching of standard LS? What's the result? will `serverMode` end up with "Standard" or "lightweight"?

Reply:
You will get standard - with a pop-up dialog to ask you reload window to make the setting change active

Reply:
got it.